x/stakeibc/keeper/host_zone.go

```diff
@@ -124,6 +125,69 @@ func (k Keeper) AddDelegationToValidator(ctx sdk.Context, hostZone types.HostZon
 	return false
 }
 
+func (k Keeper) AddValidatorToHostZone(ctx sdk.Context, msg *types.MsgAddValidator, fromGovernance bool) error {
+	// Get the corresponding host zone
+	hostZone, found := k.GetHostZone(ctx, msg.HostZone)
+	if !found {
+		errMsg := fmt.Sprintf("Host Zone (%s) not found", msg.HostZone)
+		k.Logger(ctx).Error(errMsg)
+		return sdkerrors.Wrap(types.ErrHostZoneNotFound, errMsg)
+	}
+
+	// Get max number of validators and confirm we won't exceed it
+	maxNumVals, err := cast.ToIntE(k.GetParam(ctx, types.KeySafetyNumValidators))
+	if err != nil {
+		errMsg := fmt.Sprintf("Error getting safety max num validators | err: %s", err.Error())
+		k.Logger(ctx).Error(errMsg)
+		return sdkerrors.Wrap(types.ErrMaxNumValidators, errMsg)
+	}
+	if len(hostZone.Validators) >= maxNumVals {
```

Comment:
In "maxNumVals" do we only care about validators with 0 weight? If so, we should change this check! 

The specific use-case I'm imagining is we change one validator's weight to 0 so we can add another. As far as I know, we shouldn't run into any gas issues if this is the case

Reply:
great point! We'll also want to ignore 0 weights when finding the min weight!

Reply:
addressed